README.md

```diff
@@ -1,50 +1,419 @@
+
+### `src/routes/`
+
+Manages application routing configuration.
+
+- **Purpose**: Define and organize application routes
+- **Contents**:
+  - Route definitions
+  - Route protection logic
+  - Navigation paths
+- **Key Files**:
+  - `index.tsx`: Main router configuration
+  - `paths.ts`: Centralized route constants
+  - `private-routes.ts`: Protected route definitions
+  - `public-routes.ts`: Publicly accessible routes
+
+### `src/services/`
+
+Handles external service interactions and API communications.
+
+- **Purpose**: Centralize network requests and service logic
+- **Contents**:
+  - API call implementations
+  - Authentication service
+  - External service integrations
+- **Examples**:
+  - `api.ts`: Base API configuration
+  - `auth.ts`: Authentication-related service methods
+
+### `src/stores/`
+
+Manages global state using Zustand.
+
+- **Purpose**: Provide centralized state management
+- **Contents**:
+  - Global state stores
+  - State manipulation logic
+- **Examples**:
+  - `useAuthStore.ts`: Authentication state management
+  - Other feature-specific stores
+
+## Architectural Principles
+
+1. **Modularity**: Each feature is self-contained and can be developed independently
+2. **Type Safety**: Comprehensive TypeScript usage
+3. **Separation of Concerns**: Clear boundaries between different parts of the application
+4. **Scalability**: Structure allows easy addition of new features
+
+## Getting Started
+
+### Prerequisites
+
+- Node.js (v22.0.0 or newer)
+- PNPM (v9.0.0 or newer)
+
+### Installation Steps
+
+1. Clone the repository
+2. Run `pnpm install` to install dependencies
+3. Copy `env.example` to `.env` and configure your environment variables
+4. Run `pnpm dev` to start the development server
+
+## Development Workflow
+
+### Running the Project
+
+- `pnpm dev`: Start development server
+- `pnpm build`: Create production build
+- `pnpm lint`: Run ESLint
+- `pnpm preview`: Preview production build
+
+### Code Quality Tools
+
+- **ESLint**: Ensures code quality and catches potential issues
+- **Prettier**: Maintains consistent code formatting
+- **Husky**: Runs linters before commits
+
+## State Management
+
+We use Zustand for state management in our application. This section outlines our conventions and best practices for working with Zustand stores.
+
+### Store Structure
+
+Each store should be defined in a separate file under `src/stores` directory. The store file should:
+
+1. Define an interface for the store state
+2. Create and export the store using Zustand's `create` function
+3. Include any necessary actions as part of the store
+
+Example structure:
+
+```typescript
+import { create } from 'zustand';
+import { devtools, persist } from 'zustand/middleware';
+
+interface StoreState {
+  // State properties
+  someProperty: SomeType;
+
+  // Actions
+  setSomeProperty: (value: SomeType) => void;
+}
+
+export const useStore = create<StoreState>()(
+  devtools(
+    persist(

```

Comment:
remove the persist layer. It's optional and only required in few cases.

Reply:
<img width="518" alt="image" src="https://github.com/user-attachments/assets/397159ed-7989-4a47-a70b-982226fda71d">

As it is part of the example store, I think it will serve only as an example. I also added a comment regarding the use case in the code snippet.